docs/topics/inner_workings.rst

```diff
@@ -0,0 +1,116 @@
+##############
+Inner workings
+##############
+
+Types of objects
+================
+
+Conceptually, TPV consists of the following types of objects.
+
+1. **Entities** - An entity is anything that will be considered for scheduling
+by TPV. Entities include Tools, Users, Groups, Rules and Destinations.
```

Comment:
is this true of Groups?

Reply:
By Groups, I meant Roles, so I guess it might be better to rename it?